lib/rubycritic/core/analysed_module.rb

```diff
@@ -19,6 +23,22 @@ def path
       @path ||= pathname.to_s
     end
 
+    def file_location
+      file_path[0...-1].join('/')
+    end
+
+    def file_name
+      file_path[-1]
```

Comment:
`pathname.basename` or `pathname.basename.sub_ext('').to_s` if you don't want the file extension, as seen in the `Location` class.